src/hotspot/share/prims/jvmtiThreadState.cpp

```diff
@@ -269,6 +269,7 @@ JvmtiVTMSTransitionDisabler::disable_VTMS_transitions() {
     ThreadBlockInVM tbivm(thread);
     MonitorLocker ml(JvmtiVTMSTransition_lock, Mutex::_no_safepoint_check_flag);
 
+    assert(!thread->is_in_tmp_VTMS_transition(), "sanity check");
     assert(!thread->is_in_VTMS_transition(), "VTMS_transition sanity check");
```

Comment:
Why not cover both of these with `!thread->is_in_any_VTMS_transition()`?

Reply:
It is to be clear what condition caused the assert as they are different beasts.

Reply:
ok